lib/worker.go

```diff
@@ -269,11 +282,31 @@ func (w *DefaultBaseWorker) doPostInit(ctx context.Context) error {
 }
 
 func (w *DefaultBaseWorker) doPoll(ctx context.Context) error {
-	if err := w.messageHandlerManager.CheckError(ctx); err != nil {
-		return err
+	if err := w.errCenter.CheckError(); err != nil {
```

Comment:
I notice that ErrCenter only cares about the first error.
What if we get a app error first here and trigger the suicide next, does it that we can't handle suicide?

Reply:
Resolved in the latest commit